### PR TITLE
Adds an integration test template generation script

### PR DESCRIPTION
**Pull request by cmcantalupo**
- Gets rid of copying_headers/MANIFEST.BSD3-intel entirely.
- Fixup some issues with files not getting cleaned up by integration tests.

### CODE REVIEW

**Review by bakerbrandond — COMMENTED** [review]
Took a look and would like someone else to as well

File: copying_headers/test-license [hunk 1]
```diff
@@ -210,7 +210,9 @@ class Repo(object):
 
         # Create a list of paths to partial manifest files
         self.manifest_list = [os.path.join(self.copying, path)
-                              for path in listing if path.startswith('MANIFEST.')]
+                              for path in listing
+                              if path.startswith('MANIFEST.') and
+                              path != 'MANIFEST.BSD-intel']
```
Comment (dianarg) [inline copying_headers/test-license]: MANIFEST.BSD3-intel

Comment (dianarg) [reply copying_headers/test-license]: also look into why this still worked

File: test_integration/test_ee_stream_dgemm_mix.py [hunk 2]
```diff
@@ -91,6 +91,8 @@ def setUpClass(cls):
         cls._skip_launch = _g_skip_launch
         cls._keep_files = os.getenv('GEOPM_KEEP_FILES') is not None
         cls._agent_conf_path = test_name + '-agent-config.json'
+        if 'exc_clear' in dir(sys):
```
Comment (dianarg) [inline test_integration/test_ee_stream_dgemm_mix.py]: can this be contained somewhere?

**Review by dianarg — CHANGES_REQUESTED** [review]

File: copying_headers/test-license (hunk 1 above)
Comment (cmcantalupo) [reply copying_headers/test-license]: Took a while to get to the bottom of this one, but the code is now fixed.

File: test_integration/test_ee_stream_dgemm_mix.py (hunk 2 above)
Comment (cmcantalupo) [reply test_integration/test_ee_stream_dgemm_mix.py]: Moved into geopmy.error

File: test_integration/test_template.py.in [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        sys.stdout.write('(' + os.path.basename(__file__).split('.')[0] +
+                         '.' + cls.__name__ + ') ...')
+        test_name = '@test_name@'
+        cls._report_path = test_name + '.report'
+        cls._trace_path = test_name + '.trace'
+        cls._skip_launch = _g_skip_launch
+        cls._keep_files = os.getenv('GEOPM_KEEP_FILES') is not None
+        cls._agent_conf_path = test_name + '-agent-config.json'
+        # Clear out exception record for python 2 support
+        if 'exc_clear' in dir(sys):
```
Comment (dianarg) [inline test_integration/test_template.py.in]: another one here, probably need to add an import for geopm.error

File: test_integration/test_template.py.in [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        """
+        report = geopmpy.io.RawReport(self._report_path)
+
+if __name__ == '__main__':
+    try:
+        sys.argv.remove('--skip-launch')
+        _g_skip_launch = True
+    except ValueError:
+        if 'exc_clear' in dir(sys):
+            sys.exc_clear()
```
Comment (dianarg) [inline test_integration/test_template.py.in]: another one

File: test_integration/test_template.py.in [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+        """
+
+        if (sys.exc_info() == (None, None, None) and not
+            cls._keep_files and not
+            cls._skip_launch):
+            os.unlink(cls._agent_conf_path)
+            os.unlink(cls._report_path)
+            for tf in glob.glob(cls._trace_path + '.*'):
+                os.unlink(tf)
```
Comment (dianarg) [inline test_integration/test_template.py.in:144]: it would be nice if this logic was available in a common place too (maybe not for this PR)

**Review by dianarg — CHANGES_REQUESTED** [review]

File: test_integration/test_template.py.in (hunk 5 above)
Comment (cmcantalupo) [reply test_integration/test_template.py.in:144]: Yeah, we need to have a common module for integration tests. Will not fix this now.

**Review by dianarg — APPROVED** [review]

**Pull request merged**